### PR TITLE
use gradle apt plugin

### PR DESCRIPTION
**Pull request by atomfrede**
using apt plugin to make mapstruct config for gradle a lot easier. Also update mapstruct to latets cr2.

### CODE REVIEW

File: app/templates/_gradle.properties [hunk 1]
```diff
@@ -35,7 +35,7 @@ mysql_connector_java_version=5.1.36<% } %>
 h2_version=1.4.188
 <% if (testFrameworks.indexOf('gatling') != -1) { %>
 gatling_version=2.1.7<% } %>
-mapstruct_version=1.0.0.CR1<% if (enableSocialSignIn) { %>
```
Comment (jdubois) [inline app/templates/_gradle.properties]: We have a bug with CR2, we should stick to CR1 (as with the Maven build normally)

File: app/templates/_build.gradle [hunk 2]
```diff
@@ -12,8 +12,9 @@ buildscript {
         classpath group: 'org.springframework.build.gradle', name: 'propdeps-plugin', version: '0.0.7'
         classpath group: 'com.moowork.gradle', name: 'gradle-node-plugin', version: '0.11'<% if(frontendBuilder == 'grunt') {%>
         classpath group: 'com.moowork.gradle', name: 'gradle-grunt-plugin', version: '0.11'<% }  if(frontendBuilder == 'gulp') {%>
-  	    classpath group: 'com.moowork.gradle', name: 'gradle-gulp-plugin', version: '0.11'
-        <% } %>
+  	    classpath group: 'com.moowork.gradle', name: 'gradle-gulp-plugin', version: '0.11'<% } %>
+        classpath group: 'net.ltgt.gradle', name: 'gradle-apt-plugin', version: '0.4'
```
Comment (jdubois) [inline app/templates/_build.gradle]: shouldn't this be inside the closing bracket from the line above?

File: app/templates/_gradle.properties (hunk 1 above)
Comment (atomfrede) [inline app/templates/_gradle.properties]: You're right I remember the discussion now, will revert it.

File: app/templates/_build.gradle (hunk 2 above)
Comment (atomfrede) [inline app/templates/_build.gradle]: Will have a look, but it seems correct to me. The apt plugin should always be inserted, or do I get you wrong?

**Comment by jdubois** [comment]
And of course MapStruct 1.0 was released today.... I'll first merge this, we'll upgrade later.

**Pull request merged**